### DISCUSSION TITLE
Ender3 V2/S1 Professional Firmware 20220302 - Maintenance release III

### DISCUSSION THREAD

**Discussion by mriscoc** [mriscoc/Ender3V2S1#88] [Announcements]
Universal Edition
============

This firmware is based on Marlin, has been optimized and has enhancements for the Creality Ender 3 V2/S1 printer. This is an **universal firmware** that can be used in both RET6 and RCT6 SoC chips that have 256k or 512k Flash memory. 

## Versions
- **Ender 3v2 Manual Mesh**: if you have the Ender 3v2 near to stock then use this version.
- **Ender 3v2 BLTouch**: use this version if you installed a CR/3D/BLTouch level sensor in the Ender 3v2.
- **Ender 3S1**: use this version if you have an Ender 3S1 printer

<img align="left" src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main.jpg" width="270" height="480" />
<img  src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/main-iconset.jpg" width="270" height="480" />

The binary files of this release are up to date and you can download them from the assets at the bottom of this page. To know if you have the last release go to the info page in the display of your printer and compare the release build date with the name of the binary file. The filename follows the convention:  
  
Ender3v2-422-ManualMesh-**YYYYMMDD**.bin
  
Where YYYYMMDD indicates the date of the build. 422 refers to the mainboard version **4.2.2**, and 427 is for the **4.2.7** mainboard version, if you got a **4.3.1** board use the 422 version. 

## Special versions
There are special compiles (as V6/Volcano) here: https://github.com/mriscoc/Marlin_Configurations/releases

<img src="https://cdn.shopify.com/s/files/1/0046/3781/8929/files/1-11.jpg" width="320" height="320" />

## Community links
* [Telegram](https://t.me/ender3v2s1firmware)
* [Reddit](https://www.reddit.com/r/Ender3v2Firmware) 
* [Ender 3V2 Facebook](https://www.facebook.com/groups/ender3v2firmware)
* [Ender 3S1 Facebook](https://www.facebook.com/groups/ender3s1printer)

## Donations
Thank you for your support, I receive donations through [Patreon](https://www.patreon.com/mriscoc) and [Paypal](https://www.paypal.com/paypalme/mriscoc)   

[<img src="https://www.paypalobjects.com/en_US/i/btn/btn_donateCC_LG.gif">](https://www.paypal.com/donate?business=85SPAAR6UZEE8&currency_code=USD)

# Firmware features

* Universal RCT6/RET6 Ender 3 V2/S1 Support
* Improved support for Manual Mesh and CR/3D/BLTouch
* User configurable power lost recovery enabler
* Manual and assisted bed tramming
* Tramming Wizard for BL/3D/CRTouch versions
* Long filename support with scroll
* Baby Z-Stepping
* Firmware Retract
* Advanced pause feature
* Configurable park-head position (default x=240 y=220)
* Filament change support
* Filament management (load/unload/park head) in prepare menu
* Print Flow on Motion and Tune menu 
* Home axis after motor disabling
* Fastest Z feed rate
* Live move for all motors but extruder
* Support for S-curve Acceleration
* Reboot printer item on control menu
* Increase extruder feed rate limit to 150 for BMG users
* Allows to invert direction of the extruder on Motion menu
* Support for BTT Smart filament sensor
* Selectable runout filament sensor active state (LOW/HIGH/MOTION)
* Editable runout distance
* Support for µSD extensions
* PID popup with temperature plot
* Lock screen
* Editable physical dimensions
* Mesh viewer
* Custom pre-heat
* Enable/disable beeps sound
* New print g-code preview (use [this ](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-generate-a-gcode-preview)guide)
* New print done screen
* Two decimal positions for axis coordinates value
* New Z-Offset wizard
* New PID settings menus
* PID moves the head to the center for more accuracy
* Leveling bed temperature linked to bed PID temperature
* Editable extrude minimum temperature
* Brightness settings on tune menu
* Turn off the backlight of the lcd in the tune menu
* New status line system, allows the scroll of large messages
* Home submenu with individual axis homing
* Enabled user setting for 115200 bauds (Creality Box compatibility)
* Head parking before recovery after power failure
* Offset icon change to show leveling status
* Live end-stop diagnostic page
* Move bed to front after cancel a print.
* Editable firmware retracts settings for Tune and filament settings menu
* Print Statistics page accessible from the Advanced Settings menu

![latest release](https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/New-beta.gif)

### Manual mesh
* Assisted manual mesh leveling menu
* Selectable mesh points from 2x2 to 9x9
* Mesh viewer
* Assisted banual bed tramming
* Persistent Z-offset

### CR/3D/BLTouch
* Safe Probe Z-Homing
* Bed auto level
* High speed probe
* Retry bed level on error
* Mesh viewer
* Assisted tramming
* Selectable mesh points from 2x2 to 9x9
* Allows to the user to cancel an auto leveling process 
* Increase Max Probe Offset and Probing Margin to 60 mm
* M48 Probe Test to Probe Settings menu
* Probe Stow/Deploy in Probe Settings menu
* Option to disable HS mode for better compatibility with CRTouch
* Option to change the Z Probe feed rate
* Configurable Bed auto leveling temperature

>Use the 2x2 mesh to level you bed!

<a href="https://youtu.be/qzYJC2AHbW8"><img src="https://img.youtube.com/vi/qzYJC2AHbW8/maxresdefault.jpg" width="640" height="400" /></a>  
  
### G-Codes enhanced support
* Filament Load Unload G-codes (M701/M702)
* Support for M600 filament change
* Support for G-code M117 messages
* Enable M575 change baud rate by G-code
* Extended M75 with filename support, use M75 {{ event.name }} in g-code script
* Enable ARC and 'P' parameter to specify complete circles (G2/G3)
* G35 support
* M155 Geometry report
* Native support of marlin brightness and M256 g-code

### Support for special nonstandard custom C G-Codes
* Implemented C11 En Rn Gn Bn gcode for colorize UI elements
* C108 to cancel screens waiting for user simple confirmation
* C29 N# to set amount of grid points of the mesh leveling
* C100 X### Y### to setup minimum limits
* C101 X### Y### z### to setup maximum limits
* C102 X### Y### to setup bed size
* C125 X### Y### z### to setup the park position
* C412 M# set run-out sensor active state (M0:LOW, M1:HIGH, M2:MOTION)
* C510 to lock/unlock the screen (C510 U1 to unlock)
* C562 E# to invert the Extruder (E1 to invert)
* C851 M### F#### to set probing margin, Z feed rate

<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/themes/Select_Colors.jpg" width="270" height="480" /> 

### Advanced settings menu
* Home XYZ offsets menu
* Probe offsets menu (BLTouch version)
* Customizable park-head position
* Customizable bed size and z heigh up to 999 x 999 x 999
* Launch PID for nozzle and bed with graphical feedback
* Filament run-out sensor with active state setup (LOW/HIGH)
* Filament load and unload lengths menu items in run-out menu
* Power-loss recovery enable/disable option
* LCD Brightness control
* Select Color Sub Menu
* Beep/Sound enable/disable option
* Selectable mesh point (BLTouch version)
* Mesh viewer
* Lock screen

### Special Octoprint / Host print support
* High speed (250000 baud) serial communication
* In screen control of host printing (M73, M75-M77) 
* Display messages on status line (M117)
* Color change of UI elements for visual status report (C11)
* Binary file transfer for remote firmware upgrade
* List binary (.bin) files (M20) for flash firmware support with BFT
* Arc Welder Octoprint plugin support
* EEPROM Editor Octoprint plugin support
* Host action commands for filament runout
* Emergency Parser active
* Advanced OK

**Octoprint bed visualizer support**
<img src="https://raw.githubusercontent.com/mriscoc/Ender3V2S1/Ender3V2S1-Released/screenshots/Bed-visualizer.png" width="350" height="225" />

### Display enhancements
* Extended DWIN library
* Enhanced status messages
* Improved status bar with temperature, flow, fan, z-offset and xyz positions
* PID auto tune progress on display with plot
* Leveling progress on display
* Boot screen for M997
* Transparent icons for better color theme adaptability
* New print g-code preview
* New print done screen

**A lot of additions and fixes from Bugfix development branch**

## More info on the wiki pages ##
- [**How to install the firmware**](https://github.com/mriscoc/Ender3V2S1/wiki/How-to-install-the-firmware)

- [**Octoprint settings**](https://github.com/mriscoc/Ender3V2S1/wiki/Octoprint)

- [**How to make the Manual Mesh**](https://github.com/mriscoc/Ender3V2S1/wiki/Manual-Mesh)

- [**How to install a 3D/BLTouch**](https://github.com/mriscoc/Ender3V2S1/wiki/3D-BLTouch)

- [**How to apply a color theme**](https://github.com/mriscoc/Ender3V2S1/wiki/Color-Themes)

## What's new in this release ##
* Better compatibility with DACAI screens
* Increases axes speeds
* Increases editing limits for axes speed and acceleration
* Fix incompatibility with BILINEAR_SUBDIVISION
* Fix bug: Printer hangs if g-code file has an incompatible thumbnail
* Some minor fixes
* Merge latest Marlin Bugfix

https://user-images.githubusercontent.com/2745567/143180943-f69f5967-34c1-421f-86a1-466ecccab97d.mp4

## Precompiled firmware

<hr /><em>This discussion was created from the release <a href='https://github.com/mriscoc/Ender3V2S1/releases/tag/20220314'>Ender3 V2/S1 Professional Firmware 20220302 - Maintenance release III</a>.</em>

**Comment by Stardoc70** [comment]
I updated to latest release but there is a problem in z homing, it does auto homing but it doesnt move when you do z homing and the z height stays 2 cm it doesnt go down
